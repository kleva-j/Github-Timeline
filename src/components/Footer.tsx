import React from 'react';

interface Props {
  absolute?: boolean;
}

export const Footer: React.FC<Props> = ({ absolute }) => {
  return (
    <>
      <footer
        data-testid="footer"
        className={(absolute ? 'absolute w-full bottom-0 bg-gray-900' : 'relative') + ' pb-6'}
      >
        <div className="container mx-auto px-4">
          <hr className="mb-6 border-b-1 border-gray-700" />
          <div className="flex flex-wrap items-center md:justify-between justify-center">
            <div className="w-full md:w-4/12 px-4">
              <div className="text-sm text-white font-semibold py-1">
                Copyright Â© {new Date().getFullYear()}{' '}
                <a
                  href="http://michaelobasi.com"
                  className="text-white hover:text-gray-400 text-sm font-semibold py-1"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Github Timeline
                </a>
              </div>
            </div>
            <div className="w-full md:w-8/12 px-4">
              <ul className="flex flex-wrap list-none md:justify-end  justify-center">
                <li>
                  <a
                    href="http://michaelobasi.com"
                    className="text-white hover:text-gray-400 text-sm font-semibold block py-1 px-3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Michael Obasi
                  </a>
                </li>
                <li>
                  <a
                    href="http://michaelobasi.com"
                    className="text-white hover:text-gray-400 text-sm font-semibold block py-1 px-3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    About Me
                  </a>
                </li>
                <li>
                  <a
                    href="https://frontendtips.netlify.app/"
                    className="text-white hover:text-gray-400 text-sm font-semibold block py-1 px-3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Blog
                  </a>
                </li>
                <li>
                  <a
                    href="https://github.com/kleva-j/Github-Timeline/blob/develop/LICENCE"
                    className="text-white hover:text-gray-400 text-sm font-semibold block py-1 px-3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    MIT License
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </footer>
    </>
  );
};
