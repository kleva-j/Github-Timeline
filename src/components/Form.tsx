import React from 'react';

export const SearchForms: React.FC = () => {
  return (
    <form className="flex flex-col sm:flex-row max-w-3xl mx-auto">
      <aside className="flex-1 my-2 md:mx-2 md:ml-0">
        <label htmlFor="desc" className="text-sm font-semibold">Job Description</label>
        <input
          id="desc"
          type="text"
          name="description"
          placeholder="Filter by title, benefits, companies, expertise"
          className="w-full py-2 px-4 border rounded bg-white"
        />
      </aside>

      <aside className="flex-1 my-2 sm:mx-2 md:mr-0">
        <label htmlFor="location" className="text-sm font-semibold">Location</label>
        <input
          type="text"
          id="location"
          name="location"
          placeholder="Filter by city, state, zip code or country"
          className="w-full py-2 px-4 border rounded bg-white"
        />
      </aside>
    </form>
  );
};
