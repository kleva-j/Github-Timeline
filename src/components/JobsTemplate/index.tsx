export * from './template1';
export * from './template2';
