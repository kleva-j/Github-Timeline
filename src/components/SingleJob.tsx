import React from 'react';
import PropTypes from 'prop-types';
import { useQuery } from '@apollo/client';

import { LOAD_SINGLE_JOB } from '../GraphQL/Queries';

export interface Job {
  id: string;
  title: string;
  type: string;
  company: string;
  description: string;
  url: string;
  location: string;
  created_at: string;
}

interface Props {
  id: string;
}

export const GetSingleJob: React.FC<Props> = ({ id }) => {
  const { error, loading, data } = useQuery(LOAD_SINGLE_JOB, {
    variables: { id },
  });

  if (loading) return <p data-testid="loading">Loading...</p>;
  if (error) return <p data-testid="error">Oops, Error occured!!!</p>;

  return (
    <aside>
      <div key={data?.job?.id} className="bg-primary-input mx-4 my-4">
        <h5 data-testid="title">{data?.job?.title}</h5>
        <p data-testid="company">{data?.job?.company}</p>
        <p data-testid="type">{data?.job?.type}</p>
        <p data-testid="location">{data?.job?.location}</p>
        <p data-testid="created">{data?.job?.created_at}</p>
        <a data-testid="url" href={data?.job?.url}>
          Read More...
        </a>
      </div>
    </aside>
  );
};

GetSingleJob.propTypes = {
  id: PropTypes.string.isRequired,
};
