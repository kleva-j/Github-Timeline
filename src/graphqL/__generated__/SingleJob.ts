/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: SingleJob
// ====================================================

export interface SingleJob_job_commitment {
  __typename: "Commitment";
  title: string;
  slug: string;
}

export interface SingleJob_job_cities_country {
  __typename: "Country";
  name: string;
  isoCode: string | null;
}

export interface SingleJob_job_cities {
  __typename: "City";
  name: string;
  country: SingleJob_job_cities_country;
}

export interface SingleJob_job_countries {
  __typename: "Country";
  name: string;
}

export interface SingleJob_job_remotes {
  __typename: "Remote";
  name: string;
  type: string;
  slug: string;
}

export interface SingleJob_job_company {
  __typename: "Company";
  name: string;
  slug: string;
  websiteUrl: string;
  logoUrl: string | null;
}

export interface SingleJob_job_tags {
  __typename: "Tag";
  name: string;
  slug: string;
}

export interface SingleJob_job {
  __typename: "Job";
  id: string;
  title: string;
  slug: string;
  commitment: SingleJob_job_commitment;
  cities: SingleJob_job_cities[] | null;
  countries: SingleJob_job_countries[] | null;
  remotes: SingleJob_job_remotes[] | null;
  description: string | null;
  applyUrl: string | null;
  company: SingleJob_job_company | null;
  tags: SingleJob_job_tags[] | null;
  isPublished: boolean | null;
  isFeatured: boolean | null;
  locationNames: string | null;
  userEmail: string | null;
  postedAt: any;
  createdAt: any;
  updatedAt: any;
}

export interface SingleJob {
  job: SingleJob_job;
}

export interface SingleJobVariables {
  companySlug: string;
  jobSlug: string;
}
