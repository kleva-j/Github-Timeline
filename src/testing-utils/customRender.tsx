import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';

export const waitForData = () => new Promise((res) => setTimeout(res, 0));

export const customRender = (node: JSX.Element, mocks?: MockedResponse[]) => {
  return {
    ...render(
      <MockedProvider mocks={mocks} addTypename={false}>
        {node}
      </MockedProvider>,
    ),
  };
};
