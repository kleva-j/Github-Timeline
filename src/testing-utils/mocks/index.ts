import { LOAD_JOBS, LOAD_SINGLE_JOB } from 'GraphQL/Queries';

export const mockSingleJob = {
  request: {
    query: LOAD_SINGLE_JOB,
    variables: { id: 'XXXX' },
  },
  result: {
    data: {
      job: {
        id: 'XXXX',
        title: 'Buck',
        type: 'bulldog',
        company: 'company',
        description: 'description',
        url: 'some_random_url',
        location: 'location',
        created_at: 'Fri Mar 12 09:57:21 UTC 2021',
        company_url: 'some_company_url',
        company_logo: 'some_company_logo',
        how_to_apply: 'how_to_apply',
      },
    },
  },
};

export const errorMockSingleJob = {
  request: mockSingleJob.request,
  error: new Error('An error occurred'),
};

export const mockAllJobs = {
  request: {
    query: LOAD_JOBS,
    variables: {},
  },
  result: {
    data: {
      jobs: [
        {
          id: 'XXXX',
          title: 'Buck',
          type: 'bulldog',
          company: 'company',
          description: 'description',
          url: 'some_random_url',
          location: 'location',
          created_at: 'Fri Mar 12 09:57:21 UTC 2021',
          company_url: 'some_company_url',
          company_logo: 'some_company_logo',
          how_to_apply: 'how_to_apply',
        },
      ],
    },
  },
};

export const errorMockAllJobs = {
  request: mockAllJobs.request,
  error: new Error('An error occurred'),
};
