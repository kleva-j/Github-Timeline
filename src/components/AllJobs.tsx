import React from 'react';
import { useQuery } from '@apollo/client';

import { LOAD_JOBS } from '../GraphQL/Queries';
import { Job } from './SingleJob';

export const GetAllJobs: React.FC = () => {
  const { error, loading, data } = useQuery(LOAD_JOBS);

  if (loading) return <p data-testid="loading">Loading...</p>;
  if (error) return <p data-testid="error">Oops, Error occured!!!</p>;

  return (
    <section className="grid__section min-h-screen bg-primary-bg">
      <h2>Github Jobs Listings</h2>
      <article className="flex flex-wrap justify-evenly">
        {data?.jobs.map(({ id, title, type, company, url, location, created_at }: Job) => (
          <aside data-testid={id} key={id} className="bg-primary-input mx-4 my-4">
            <h5 data-testid="title">{title}</h5>
            <p data-testid="company">{company}</p>
            <p data-testid="type">{type}</p>
            <p data-testid="location">{location}</p>
            <p>{created_at}</p>
            <a href={url}>Read More...</a>
          </aside>
        ))}
      </article>
    </section>
  );
};
