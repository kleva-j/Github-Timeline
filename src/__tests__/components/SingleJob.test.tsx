import { cleanup } from '@testing-library/react';

import { GetSingleJob } from 'components/SingleJob';
import { waitForData, customRender } from 'testing-utils/customRender';
import { errorMockSingleJob, mockSingleJob } from 'testing-utils/mocks';

describe('GetSingleJob', () => {
  afterEach(cleanup);

  test('it should test the loading state.', async () => {
    const { getByTestId } = customRender(<GetSingleJob id="XXXX" />, [mockSingleJob]);
    const loading = getByTestId('loading');
    expect(loading.textContent).toContain('Loading...');
  });

  test('it should test the success state.', async () => {
    const { container, getByTestId, getByText } = customRender(<GetSingleJob id="XXXX" />, [mockSingleJob]);
    await waitForData();
    expect(container.firstChild).toMatchSnapshot();
    expect(getByText('Read More...')).toBeTruthy();
    const url = getByTestId('url');
    expect(url.textContent).toContain('Read More...');
  });

  test('it should test the error state.', async () => {
    const { getByTestId } = customRender(<GetSingleJob id="XXXX" />, [errorMockSingleJob]);
    await waitForData();
    const error = getByTestId('error');
    expect(error.textContent).toContain('Oops, Error occured!!!');
  });
});
