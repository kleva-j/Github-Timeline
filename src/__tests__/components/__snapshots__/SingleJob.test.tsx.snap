// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetSingleJob it should test the success state. 1`] = `
<aside>
  <div
    class="bg-primary-input mx-4 my-4"
  >
    <h5
      data-testid="title"
    >
      Buck
    </h5>
    <p
      data-testid="company"
    >
      company
    </p>
    <p
      data-testid="type"
    >
      bulldog
    </p>
    <p
      data-testid="location"
    >
      location
    </p>
    <p
      data-testid="created"
    >
      Fri Mar 12 09:57:21 UTC 2021
    </p>
    <a
      data-testid="url"
      href="some_random_url"
    >
      Read More...
    </a>
  </div>
</aside>
`;
